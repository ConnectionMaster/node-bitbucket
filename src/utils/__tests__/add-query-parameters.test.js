const addQueryParameters = require('../add-query-parameters')

const url = '/'

describe('utils:add-query-parameters', () => {
  it('returns url as is when params is empty', () => {
    expect(addQueryParameters(url)).toBe(url)
    expect(addQueryParameters(url, {})).toBe(url)
  })

  it('works on fresh url', () => {
    expect(addQueryParameters(url, { wibbly: 'wobbly' })).toMatchSnapshot()
  })

  it('works on url containing query string', () => {
    let urlWithQS = addQueryParameters(url, { wibbly: 'wobbly' })
    expect(addQueryParameters(urlWithQS, { timey: 'wimey' })).toMatchSnapshot()
  })

  it('encodes uri components', () => {
    expect(addQueryParameters(url, { query: '&+' })).not.toMatch(/&|\+/)
  })

  it('does not encode `+` when query name is `q`', () => {
    expect(addQueryParameters(url, { q: 'doctor+who?' })).toMatch(/\+/)
  })
})
