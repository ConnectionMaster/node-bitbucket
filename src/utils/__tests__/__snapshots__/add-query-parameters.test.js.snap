// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils:add-query-parameters works on fresh url 1`] = `"/?wibbly=wobbly"`;

exports[`utils:add-query-parameters works on url containing query string 1`] = `"/?wibbly=wobbly&timey=wimey"`;
